lib/trie.js

```diff
@@ -15,6 +15,9 @@ function Node (options) {
   // stores all direct folder descendants of this node
   this.children = {};
 
+  // number of folders in the sub-trie of this node
+  this.childCount = options ? options.childCount : 0;
```

Comment:
@Dhroov7 Are we ever creating a Node with childCount != 0?
We're already specifying `children={}`. How can childCount be anything but 0?